src/js/map/boundary_type_box.js

```diff
@@ -23,69 +23,90 @@ export class BoundaryTypeBox extends Component {
         this._activeVersion = value;
     }
 
-    activeVersionUpdated = (activeVersion) => {
+    activeVersionUpdated = (geometries, activeVersion, selectedLevel) => {
         this.activeVersion = activeVersion;
+        let levels = Object.keys(geometries.children) || [];
 
-        let text = $(selectedOption).find('.dropdown-menu__selected-item .truncate').text();
-        let arr = text.split('/');
-        let childType = arr.length > 1 ? arr[1].trim() : '';
-        const selectedText = arr.length > 1 ? `${activeVersion.model.name} / ${childType}` : activeVersion.model.name;
-
-        this.setSelectedOption(selectedText);
+        if(levels.length > 0){
+          const selectedText = `${activeVersion.model.name} / ${levels.includes(selectedLevel) ? selectedLevel : levels[0]}`;
+          this.setSelectedOption(selectedText);
+        }
     }
 
-    setVisibilityOfDropdown = (boundaryTypes) => {
-        if (boundaryTypes.length <= 1) {
-            $(selectElement).addClass('hidden');
+    setVisibilityOfDropdown = (versionOptions) => {
+        if (versionOptions.length === 0) {
+            $(this.selectElement).addClass('hidden');
+        } else if (versionOptions.length === 1) {
+          $(this.selectElement).find(".dropdown-menu").css('pointer-events', 'none');
+          $(this.selectElement).find(".dropdown-menu__icon").addClass("hidden");
         } else {
-            $(selectElement).removeClass('hidden');
+            $(this.selectElement).removeClass('hidden');
+            $(this.selectElement).find(".dropdown-menu").css('pointer-events', 'auto');
+            $(this.selectElement).find(".dropdown-menu__icon").removeClass("hidden");
         }
```

Comment:
So this does two things right. Show/hide the geo select and also enable disable it, which is why we have to show/hide the element as well as the icon